test/auth_plug_test.exs

```diff
@@ -8,73 +8,100 @@ defmodule AuthPlugTest do
     assert AuthPlug.init(%{}) == %{}
   end
 
-  test "Plug return 401 wiht not Authorization Header" do
-    conn = AuthPlug.call(conn(:get, "/admin"), @opts)
-
-    # redirect when auth fails
-    assert conn.status == 302
-  end
-
-  test "Plug return 401 wiht incorrect jwt header" do
-    conn =
-      conn(:get, "/admin")
-      |> put_req_header("authorization", "Bearer incorrect.jwt")
-      |> AuthPlug.call(@opts)
-
-    # redirect when auth fails
-    assert conn.status == 302
-  end
-
-  test "Fail when authorization header token is invalid" do
-    conn =
-      conn(:get, "/admin")
-      |> put_req_header("authorization", "Bearer this.will.fail")
-      |> AuthPlug.call(@opts)
-
-    # redirect when auth fails
-    assert conn.status == 302
-  end
-
-  test "Conn.assign decoded (the verified JWT)" do
-    data = %{email: "alex@dwyl.com", name: "Alex"}
-    jwt = Token.generate_jwt!(data)
-
-    conn =
-      conn(:get, "/admin")
-      |> assign(:jwt, jwt)
-      |> AuthPlug.call(%{})
-
-    assert conn.assigns.person.email == "alex@dwyl.com"
-  end
-
-  test "get_session(conn, :jwt)" do
-    data = %{email: "alice@dwyl.com", name: "Alice"}
-    jwt = Token.generate_jwt!(data)
-
-    conn =
-      conn(:get, "/admin")
-      |> AuthPlug.setup_session()
-      |> put_session(:jwt, jwt)
-      |> AuthPlug.call(@opts)
-
-    token = get_session(conn, :jwt)
-    {:ok, decoded} = AuthPlug.Token.verify_jwt(token)
-    assert Map.get(decoded, "email") == "alice@dwyl.com"
-  end
-
-  test "Plug assigns person=jwt to conn with valid jwt" do
-    data = %{email: "person@dwyl.com", session: 1}
-    jwt = Token.generate_jwt!(data)
-
-    conn =
-      conn(:get, "/admin")
-      |> put_req_header("authorization", "Bearer #{jwt}")
-      |> AuthPlug.call(%{})
-
-    token = conn.assigns.jwt
-    person = AuthPlug.Token.verify_jwt!(token)
-
-    assert person["email"] == "person@dwyl.com"
+  describe "test admin endpoint" do
+    setup %{endpoint: endpoint} do
+      test_conn =
+        conn(:get, "endpoint")
+        |> init_test_session(%{})
+
+      {:ok, conn: test_conn}
+    end
+
+    @tag endpoint: "/admin"
```

Comment:
could this `@tag endpoint: "/admin"` be a module attribute instead of repeating it each time?